src/Core/Content/Category/Event/CategoryLevelLoaderCacheKeyEvent.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Category\Event;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('discovery')]
+class CategoryLevelLoaderCacheKeyEvent extends Event
+{
+    private bool $disableCaching = false;
+
+    /**
+     * @param array<mixed> $parts
+     */
+    public function __construct(
+        private array $parts,
+        private string $rootId,
+        private int $depth,
+        private SalesChannelContext $context,
+        private Criteria $criteria
+    ) {
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    public function getParts(): array
+    {
+        return $this->parts;
+    }
+
+    public function getContext(): SalesChannelContext
```

Comment:
please implement `\Shopware\Core\Framework\Event\ShopwareSalesChannelEvent` so we are consistent in naming those getters